### PR TITLE
Improve client documentation (1)

### PR DESCRIPTION
**Pull request by sascha11110**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I documented the TypeScript code using JSDoc style.

### Motivation and Context
There are many places in the client code which lack sufficient documentation. This PR is an ongoing effort to collaboratively improve the client documentation.

- [x] Only changes to the documentation were made
- [x] Documentation was made in accordance with [JSDoc style](https://jsdoc.app/about-getting-started.html)

### CODE REVIEW

**Comment by krusche** [comment]
@sascha11110 can you merge develop and resolve the conflicts before too many conflicts occur? Thanks!

**Comment by sascha11110** [comment]
I reviewed all files and made sure that only the documentation was changed (or at least only uncritical changes were made). I reverted one change on the production code here: https://github.com/ls1intum/Artemis/pull/1365/commits/de4a0ad40535b5c442480bfdf447b37d36c2b099

I also disabled the `completed-docs` tslint-rule again so that the Bamboo *TypeScript Code Style* job does not fail on develop.

@krusche So, I think we can merge this. After that I will create the second PR.

I couldn't approve the PR as I'm the creator of it.

**Comment by krusche** [comment]
@madwau can you fix the 2 merge conflicts please?

File: package.json [hunk 1]
```diff
@@ -180,7 +180,7 @@
     "scripts": {
         "prettier:format": "prettier --write 'src/{main/webapp,test}/**/*.{json,ts,js,css,scss,html}'",
         "prettier:check": "prettier --check 'src/{main/webapp,test}/**/*.{json,ts,js,css,scss,html}'",
-        "lint": "eslint . --ext .ts",
+        "lint": "eslint . --ext .ts && yarn run lint:doc",
```
Comment (krusche) [inline package.json:183]: I would like to revert this change for develop for now and rather activate one additional build job

Comment (krusche) [reply package.json:183]:
```suggestion
        "lint": "eslint . --ext .ts",
```
